### ISSUE TITLE
resolveType seems broken in v0.5.0

### ISSUE THREAD

**Issue by mariuslundgard** [graphql/graphql-js#372]
I'm using interfaces and `resolveType` in order to get lists of mixed types (following this example: https://github.com/graphql/graphql-js/blob/master/src/__tests__/starWarsSchema.js).

In v0.4.18 this works fine, however in v0.5.0 I get an error:

```
TypeError: Cannot read property 'reduce' of undefined
    at GraphQLSchema.isPossibleType (~/node_modules/graphql/type/schema.js:176:81)
    at completeAbstractValue (~/node_modules/graphql/execution/execute.js:620:26)
    at completeValue (~/node_modules/graphql/execution/execute.js:566:12)
    at completeValueCatchingError (~/node_modules/graphql/execution/execute.js:477:21)
    at ~/node_modules/graphql/execution/execute.js:590:25
    at Array.map (native)
    at completeListValue (~/node_modules/graphql/execution/execute.js:589:33)
    at completeValue (~/node_modules/graphql/execution/execute.js:554:12)
    at completeValueCatchingError (~/node_modules/graphql/execution/execute.js:477:21)
    at resolveField (~/node_modules/graphql/execution/execute.js:450:10)
```

Did the API change, or is this a bug?

**Comment by robzhu** [comment]
There was an API change: v0.5.0 introduced a reordering of arguments in the resolve function. Please take a look at this [commit](https://github.com/graphql/graphql-js/commit/d7cc6f9aed462588291bc821238650c98ad53580).

**Comment by mariuslundgard** [comment]
Ah, but it still seems to not work properly. I'm only using the first argument of `resolveType`.

Here's the code that breaks for me (with naming changed):

```js
export const SubscriptionInterface = new GraphQLInterfaceType({
  name: 'Subscription',
  fields: () => ({
    type: {
      type: GraphQLString,
      description: 'The type of the subscription.'
    }
  }),
  resolveType: (data) => {
    switch (data.type) {
      case 'email':
        return EmailSubscriptionType

      case 'notification':
        return NotificationSubscriptionType

      case 'slack':
        return SlackSubscriptionType

      default:
        return DefaultSubscriptionType
    }
  }
})
```

**Comment by helfer** [comment]
This is the line in question (176):
```js
possibleTypeMap[abstractType.name] = this.getPossibleTypes(abstractType).reduce(function (map, type) {
```
From a quick glance, it seems that there are no possible types implementing the SubscriptionInterface in your schema, which is probably because you're not providing them to your schema. This was [another breaking change](https://github.com/graphql/graphql-js/commit/6a1f23e1f9c1e6bf4cea837bc9bb6eae0fb5c382) included with 0.5.0

You need to provide the implementing types to your schema in the `types` argument if they are not otherwise reachable:
```
const schema = new GraphQLSchema({
  query: ...,
  types: [NotificationSubscriptionType, SlackSubscriptionType, DefaultSubscriptionType],
});
```

It's not a bug with GraphQL-JS, but the error message on that one should probably be improved ...

**Comment by mariuslundgard** [comment]
Thanks @helfer, providing the `types` array worked.
